yarn-project/sequencer-client/src/sequencer/index.ts

```diff
@@ -78,7 +78,10 @@ export class Sequencer {
       // Skip nullifier if it's empty
       if (nullifier.isZero()) continue;
       // TODO(AD): this is an exhaustive search currently
-      if ((await this.worldState.findLeafIndex(MerkleTreeId.NULLIFIER_TREE, nullifier.toBuffer())) !== undefined) {
+      if (
+        (await this.worldState.getLatest().findLeafIndex(MerkleTreeId.NULLIFIER_TREE, nullifier.toBuffer())) !==
```

Comment:
@PhilWindle I understand txDoubleSpend check should use latest, but please confirm.

Reply:
Theoretically it shouldn't matter. At the point of validating the new set of txs the sequencer shouldn't have started modifying any of the trees. Using getLatest is fine.